Makefile

```diff
@@ -22,7 +22,7 @@ compile: compile-client compile-server compile-csi-proxy-api-gen
 
 .PHONY: compile-client
 compile-client:
-	$(GO_ENV_VARS) go build ./client/...
+	cd client && $(GO_ENV_VARS) go build ./...
```

Comment:
Sorry this is a little unrelated but I missed the recent changes to the Makefile. Do we want to test compilation and run unit tests as part of the pull job? If so, we need a "make test" rule here.  By moving away from release-tools/build.make, we also miss some rules for verifying gofmt and vendor too

Reply:
Agree. @wk8 do you think the Makefile was csi-proxy can be reconciled with the make targets under https://github.com/kubernetes-csi/csi-proxy/blob/master/release-tools/build.make? I.e. the common items can be driven by build.make while the special requirements like protobuf gen, code gen, etc. can be driven by the top level Makefile.

I think ^ should be handled in a separate PR.